src/components/Filter.js

```diff
@@ -0,0 +1,137 @@
+} from "@/src/components/styles/FilterStyled";
+import Checkbox from "@/src/components/Checkbox";
+import { CheckboxGroup } from "@/src/components/styles/CheckboxStyled";
+import DateFilter from "@/src/components/DateFilter";
+
+export default function Filter({ filterValues, onFilterChange, onClose }) {
+  const [selectedFilters, setSelectedFilters] = useState(filterValues);
+  const filterRef = useRef(null);
+
+  useEffect(() => {
+    setSelectedFilters(filterValues);
+  }, [filterValues]);
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (filterRef.current && !filterRef.current.contains(event.target)) {
+        onClose();
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [onClose]);
+
+  const handleCheckboxChange = (filterType, value) => {
+    setSelectedFilters((prevFilters) => ({
+      ...prevFilters,
+      [filterType]: prevFilters[filterType].includes(value)
+        ? prevFilters[filterType].filter((item) => item !== value)
+        : [...prevFilters[filterType], value],
+    }));
+  };
+
+  const handleDateChange = (field, value) => {
+    setSelectedFilters((prevFilters) => ({
+      ...prevFilters,
+      [field]: value,
+    }));
+  };
+
+  const handleApplyFilter = () => {
+    onFilterChange(selectedFilters);
+    onClose();
+  };
+
+  const handleClearAll = () => {
+    const clearedFilters = {
+      transactionType: [],
+      category: [],
+      paymentMethod: [],
+      dateFrom: "",
+      dateUntil: "",
+    };
+    setSelectedFilters(clearedFilters);
+    onFilterChange(clearedFilters);
+    onClose();
+  };
+
+  return (
+    <Overlay>
+      <FilterContainer ref={filterRef}>
+        <h2>Filter transactions</h2>
+        <FilterSection>
+          <DateFilter
+            dateFrom={selectedFilters.dateFrom}
+            dateUntil={selectedFilters.dateUntil}
+            onDateChange={handleDateChange}
+          />
+        </FilterSection>
+        <FilterSection>
+          <h4>Transaction type</h4>
+          <CheckboxGroup>
+            <label>
+              <Checkbox
+                value="Expense"
+                checked={selectedFilters.transactionType.includes("Expense")}
+                onChange={() => handleCheckboxChange("transactionType", "Expense")}
+              />
+              Expense
+            </label>
+            <label>
+              <Checkbox
+                value="Income"
+                checked={selectedFilters.transactionType.includes("Income")}
+                onChange={() => handleCheckboxChange("transactionType", "Income")}
+              />
+              Income
+            </label>
+          </CheckboxGroup>
+        </FilterSection>
+        <FilterSection>
+          <h4>Categories</h4>
+          <CheckboxGroup>
+            {["Entertainment", "Food", "Shopping", "Transport"].map((category) => (
```

Comment:
Just a hint: these do not match your existing categories, maybe you wanna match them, so its easier to test your filter functionality.

Reply:
the determined categories will be implemented with the next ticket